core-web/src/main/java/org/silverpeas/core/web/session/SessionManager.java

```diff
@@ -573,6 +553,30 @@ private void openSession(SessionInfo sessionInfo) {
     }
   }
 
+  @Nonnull
+  private SessionInfo createUserSessionInfo(final User user, final HttpServletRequest request) {
```

Comment:
I thought that no JSESSIONID should have been created when createUserSessionInfo is called from `openOneShotSession` method.
A new override of SessionInfo, `OneShotSessionInfo` for example could have be implemented (using UUID as session id).
In case a normal session is open, `new HTTPSessionInfo(session, anIP, user);` is performed.
In case a one shot is open, `new OneShotSessionInfo(anIP, user)` is performed (without calling `HttpSession session = request.getSession();` instruction).

Reply:
Because, in term I want SessionInfo to be a decorator of HttpSession and merge SessionInfo and HTTPSessionInfo for mainly two reasons:

- to avoid useless complexity
- and to profit the support of Wildfly in sessions sharing among several Willdfly instances in a load-balancing infra

Beside that, the Http session is already created by Wildfly at this step. But, I'm going to check this assertion.